src/bap_mc/bap_mc.ml

```diff
@@ -170,6 +199,20 @@ module Cmdline = struct
     Arg.(value & opt_all ~vopt:`bil (enum formats) [] &
          info ["show-bil"] ~doc)
 
+  let addr =
+    let doc = "Specify an address of first byte, as though \
+	       the instructions occur at a certain address, \
+	       and accordingly interpreted. " in
+    Arg.(value & opt ~vopt:"0" (string) "0" &
+	 info ["addr"] ~doc)    
+
+  let max_insns =
+    let doc = "Specify a number of instructions to disassemble.\
+	       Good for ensuring that only one instruction is ever\
+	       lifted or disassembled from a byte blob. Default is all" in
+    Arg.(value & opt ~vopt:(Some 0) (some int) (Some 0) &
```

Comment:
The same abuse of `vopt` parameter. And using `Some 0` as a special value for infinity is actually the same as using just `0`. The idea was to use `None` value for infinity, and `Some n` for some finite `n`. Here is a better way to do things:
```ocaml
Arg.(value & opt  (some int) None & info ["max-insns"] ~doc)
```
Of course you need to update the code, that handle this parameter.